src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs

```diff
@@ -173,6 +181,7 @@ internal void LaunchTour(string tourName)
             {
                 Initialize();
                 GuideFlowEvents.OnGuidedTourStart(tourName);
+                currentGuideState = GUIDE_STATE.PLAYING;
```

Comment:
We have a list of event already, the design of states and setting it seems a bit duplicate to me

Reply:
@QilongTang I removed all the states functionality and now is using the events but I had to add a new event named OnGuidedTourClosed that will happen only for the Packages guide, due that the OnGuidedTourFinish is raised when we press the close button in the Popup but for the Packages guide we have an exit window allowing the user to continue or exit, then the OnGuidedTourClosed will be raised when the user press the Exit button, let me know how it sounds to you?
https://github.com/DynamoDS/Dynamo/pull/13002/commits/d382ee7fbcf6a3ccf04ffe5477e07e8d13d73fd2